src/Raven.Server/Documents/Sharding/ShardedDocumentDatabase.cs

```diff
@@ -226,6 +238,31 @@ public async Task DeleteBucketAsync(int bucket, long migrationIndex, string upto
         }
     }
 
+    private async Task WaitForConfirmationIndex(long confirmationIndex)
+    {
+        var cmd = new WaitForIndexNotificationCommand(new List<long> { confirmationIndex });
+        var tasks = new List<Task>(ShardingConfiguration.Orchestrator.Topology.Members.Count);
+        var clusterTopology = ServerStore.GetClusterTopology();
+
+        foreach (var nodeTag in ShardingConfiguration.Orchestrator.Topology.Members)
+        {
+            var chosenNode = new ServerNode
+            {
+                ClusterTag = nodeTag,
+                Database = ShardedDatabaseName,
+                ServerRole = ServerNode.Role.Member,
+                Url = clusterTopology.GetUrlFromTag(nodeTag)
+            };
+
+            var releaseCtx = ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context);
+            var t = ServerStore.ClusterRequestExecutor.ExecuteAsync(chosenNode, nodeIndex: null, context, cmd, token: DatabaseShutdown)
+                .ContinueWith(_ => releaseCtx.Dispose());
+            tasks.Add(t);
+        }
+
+        await Task.WhenAll(tasks);
```

Comment:
I'm worried that this will produce a very generic exception. Shouldn't we wrap it in a more meaningful one that indicates the nodeTag etc?

Reply:
done, wrapped this in `try catch` and added some info (index + orchestrator topology)
but it's still a pretty generic exception..
would you prefer if we'll await each task separately and throw a more meaningful exception (node tag of the node we failed on etc.) ?